src/poolboy.erl

```diff
@@ -415,3 +405,12 @@ state_name(#state{overflow = MaxOverflow, max_overflow = MaxOverflow}) ->
     full;
 state_name(_State) ->
     overflow.
+
+remove_from_idle(Pid, IdleWorkers) ->
+    case maps:get(Pid, IdleWorkers, undefined) of
+        undefined ->
+            IdleWorkers;
+        Timer ->
+            erlang:cancel_timer(Timer),
+            maps:remove(Pid, IdleWorkers)
+    end.
```

Comment:
as an option to read the map once

```suggestion
    case maps:take(Pid, IdleWorkers) of
        error ->
            IdleWorkers;
        {Timer, NewIdleWorkers} ->
            erlang:cancel_timer(Timer),
            NewIdleWorkers
    end.
```

Reply:
lol I searched for this function (pop) but didn't find it